cmip6_downscaling/data/observations.py

```diff
@@ -34,112 +32,76 @@ def open_era5(var):
     """
+    stores = [
+        f'https://cmip6downscaling.blob.core.windows.net/cmip6/ERA5_daily/{y}'
+        for y in range(1979, 2021)
+    ]
+    ds = xr.open_mfdataset(stores, engine='zarr', consolidated=True)
+    return ds[[var]]
+
+
+def load_obs(obs_id: str, variable: str, time_period: slice) -> xr.Dataset:
+    """Load a temporal subset of an observational dataset for one variable
     Parameters
     ----------
-    obs_id : [type]
-        [description]
-    variable : [type]
-        [description]
-    time_period : [type]
-        [description]
-    domain : [type]
-        [description]
+    obs_id : str
+        name of observation dataset. currently only "ERA5" is supported
+    variable : str
+        name of variable following CMIP conventions (e.g. tasmax, tasmin, pr)
+    time_period : slice
+        time period you want to subset. e.g. slice('1985', '2015'). using
+        full years is recommended if you are going to be doing the spatial
+        anomaly calculations.
 
     Returns
     -------
-    [xarray dataset]
-        [Chunked {time:-1,lat=10,lon=10}]
+    open_era5(variable).sel(time=time_period) : xr.Dataset
+        Temporal subset of observational dataset
     """
     ## most of this can be deleted once new ERA5 dataset complete
     ## we'll instead just want to have something like
     ## open_era5(chunking_method='space', variables=['tasmax', 'pr'], time_period=slice(start, end), domain=slice())
     if obs_id == "ERA5":
-        print("open era5")
-        full_obs = open_era5(variable)
-        print("resample era5")
-        obs = (
-            full_obs[variable_name_dict[variable]]
-            .sel(time=time_period)
-            .resample(time="1D")
-            .reduce(agg_func)
-            .rename(variable)
-            # .load(scheduler="threads")  # GOAL! REMOVE THE `LOAD`!
-        )
-    return obs
-
-
-def get_coarse_obs(
-    obs,
-    gcm_ds_single_time_slice,
-):
-    """[summary]
-
-    Parameters
-    ----------
-    obs : xarray dataset
-        chunked in space (lat=-1, lon=-1, time=1)
-    gcm_ds_single_time_slice : [type]
-        Chunked in space
-
-    Returns
-    -------
-    [type]
-        [Chunked in space (lat=-1, lon=-1, time=1)]
-    """
-    # TEST TODO: Check that obs is chunked appropriately and throw error if not
-    # Like: assert obs.chunks == (lat=-1, lon=-1, time=1) - then eventually we can move the rechunker in as an `else`
-    regridder = xe.Regridder(obs, gcm_ds_single_time_slice, "bilinear")
-
-    obs_coarse = regridder(obs)
-    return obs_coarse
+        return open_era5(variable).sel(time=time_period)
 
 
-def get_spatial_anomolies(coarse_obs, fine_obs):
-    # check if this has been done, if do the math
-    # if it has been done, just read them in
-    """[summary]
+def get_spatial_anomolies(coarse_obs, fine_obs, variable, connection_string) -> xr.Dataset:
+    """Calculate the seasonal cycle (12 timesteps) spatial anomaly associated
```

Comment:
For doc strings with longer descriptions (like this one), its best to start with a 1-line summary then follow with a longer paragraph description.

So something like this:

```
"""
Calculate the seasonal cycle (12 timesteps) spatial anomaly

The spatial anomaly is calculated by  aggregating the fine_obs to a given coarsened scale and 
then reinterpolating it back to the original spatial resolution. The outputs of this function are
dependent on three parameters:
    * a grid (as opposed to a specific GCM since some GCMs run on the same grid)
    * the time period which fine_obs (and by construct coarse_obs) cover
    * the variable
"""